operator/v1/types_ingress.go

```diff
@@ -282,6 +287,178 @@ var (
+	//
+	// If unset, the IntermediateSecurityProfile type is used.
+	//
+	// +unionDiscriminator
+	// +optional
+	Type SecurityProfileType
+	// old is the "old" backward compatibility configuration as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	//
+	// +optional
+	// +nullable
+	Old *OldSecurityProfile
+	// intermediate is the "Intermediate" compatibility configuration (default) as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	// +nullable
+	Intermediate *IntermediateSecurityProfile
+	// modern is the "Modern" compatiibility configuration as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile
+	// custom defines the scehma of a user-defined custom compatiibility configuration.
+	//
+	// +optional
+	// +nullable
+	Custom *CustomSecurityProfile
+}
+
+type OldSecurityProfile struct{}
+type IntermediateSecurityProfile struct{}
+type ModernSecurityProfile struct{}
+
+type CustomSecurityProfile struct {
+	// securityProfile defines the schema for a custom security profile.
+	//
+	// Any unspecified fields will default to the "Intermediate" compatibility configuration
+	// as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	SecurityProfile // inline
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
+
+const (
+	// SecurityProfileOldType is a security profile that maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// SecurityProfileIntermediateType is a security profile that maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// SecurityProfileModernType is a security profile that maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+	// SecurityProfileCustomType is a security profile that allows for custom settings
+	// through a SecurityProfile type.
+	SecurityProfileCustomType SecurityProfileType = "Custom"
+)
+
+// SecurityProfile defines the schema for a security profile.
+type SecurityProfile struct {
+	// ciphers is used to specify the cipher algorithms that are negotiated
+	// during the SSL/TLS handshake with an IngressController. Each cipher must
+	// be an explicit, colon-delimited list of ciphers.
+	//
+	// If unset, the "Intermediate" Ciphersuites [1] are used:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	Ciphers *string `json:"ciphers,omitempty"`
```

Comment:
Why a string instead of a list?

Reply:
I vaguely remember a conversation with @knobunc and @Miciah regarding this concern. @knobunc I believe you had a concern about using a list to specify each cipher individually. Can you confirm? If using a list is not an issue, I'll change the type.

Reply:
Yes, precisely that.  My bet is that the company sends around a string that their security team told them to use.  Also, all the security guides on the net talk about comma separated strings.  I think making it a list will be more of a pain than a benefit.

Reply:
I'm still hearing that the underlying data type is a set. How will we validate the this field? What is the schema? CSV? (Is "CSV" detailed enough here?) Doesn't that mean every client needs to parse the field using whatever embedded format we define?

In the "I have a string already" scenario above, I think it's reasonable to expect to have to adapt that into the API I'm using (i.e. convert the data to a list).

Reply:
Where I'm coming from is making sure the API caters to the broadest possible set of machine clients with minimal effort, and let clients deal with how to present a long list (31 entries in one case).

Reply:
@knobunc I updated ciphers to be a list based on the above feedback from @ironcladlou.